Cannot use `it` var on array init field as sourse array index

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** V 0.3.0 1c6366e.3ad22eb
**OS:** macos, macOS, 12.5.1, 21G83

<!-- Please include the standalone minimal reproducible code.
If not, it may be closed by the administrators. -->
**What did you do?**

```v
fn main() {
	input := [3.1, 1.1]
	arr := []f64{len: 3, init: input[it] or {0.0}}
	assert arr[0] == 3.1
	assert arr[1] == 1.1
	assert arr[2] == 0.0
}
```

**What did you expect to see?**
nothing
 
**What did you see instead?**
```
==================
   ^~~~~~~~~~~~~~~
/tmp/v_502/array_init_it.9106295923730884668.tmp.c:12100:50: error: use of undeclared identifier 'it'
                        f64* _t3 = (f64*)(array_get_with_check(input, it));
                                                                      ^
1 warning and 1 error generated.
...
==================
(Use `v -cg` to print the entire error message)
```

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -246,6 +246,7 @@ fn (mut g Gen) array_init_with_fields(node ast.ArrayInit, elem_type Type, is_amp
 		g.indent++
 		g.writeln('$elem_typ* pelem = ($elem_typ*)${tmp}.data;')
 		g.writeln('for(int it=0; it<${tmp}.len; it++, pelem++) {')
+		g.set_current_pos_as_last_stmt_pos()
 		g.indent++
 		g.write('*pelem = ')
 		g.expr(node.default_expr)
@@ -254,6 +255,7 @@ fn (mut g Gen) array_init_with_fields(node ast.ArrayInit, elem_type Type, is_amp
 		g.writeln('}')
 		g.indent--
 		g.writeln('}')
+		g.set_current_pos_as_last_stmt_pos()
 		if var_name.len == 0 {
 			if s_ends_with_ln {
 				g.writeln(s)

--- a/vlib/v/tests/array_with_it_test.v
+++ b/vlib/v/tests/array_with_it_test.v
@@ -15,3 +15,12 @@ fn test_array_with_it() {
 	a4 := []int{len: 5, init: 5 - it}
 	assert a4 == [5, 4, 3, 2, 1]
 }
+
+fn test_array_init_with_optional() {
+	input := [3.1, 1.1]
+	arr := []f64{len: 3, init: input[it] or { 0.0 }}
+	println(arr)
+	assert arr[0] == 3.1
+	assert arr[1] == 1.1
+	assert arr[2] == 0.0
+}

